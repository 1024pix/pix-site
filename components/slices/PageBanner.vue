<template>
  <div :id="`slice-${indexForId}`" :style="[background]">
    <section class="banner row-block">
      <div class="row-block__main-content">
        <prismic-rich-text
          :field="title"
          class="title--big"
          :class="`title--${fontContrast}`"
        />
        <prismic-rich-text
          :field="textContent"
          class="text--normal"
          :class="`text--${fontContrast}`"
        />
        <slot></slot>
        <div class="banner__button-group">
          <div v-for="(link, index) in links" :key="`item-${index}`">
            <nuxt-link
              v-if="!isVideo(link)"
              :to="link.banner_link_url"
              class="button"
              :class="videoClass(link)"
            >
              {{ link.banner_link_text }}
            </nuxt-link>
            <template v-if="isVideo(link)">
              <button
                class="button button-video"
                :class="`button--${fontContrast}`"
                @click="openVideoModal()"
              >
                {{ link.banner_link_text }}
              </button>
              <modal ref="modal" name="videoModal" height="auto">
                <media-player :video-url="link.banner_link_url.url" />
              </modal>
            </template>
          </div>
        </div>
      </div>
      <pix-image
        v-if="hasImage"
        :field="imageUrl"
        class="row-block__side-content"
      />
    </section>
  </div>
</template>

<script>
export default {
  name: "SlicesPageBanner",
  props: {
    slice: {
      type: Object,
      default: null,
    },
    indexForId: {
      type: Number,
      default: 0,
    },
  },
  computed: {
    fontContrast() {
      return this.slice.primary.banner_font_contrast;
    },
    title() {
      return this.slice.primary.banner_title;
    },
    textContent() {
      return this.slice.primary.banner_content;
    },
    links() {
      return this.slice.items;
    },
    hasImage() {
      return (
        this.slice.primary.banner_image && this.slice.primary.banner_image.url
      );
    },
    hasBackgroundImage() {
      return (
        this.slice.primary.banner_background &&
        this.slice.primary.banner_background.url
      );
    },
    imageUrl() {
      return this.slice.primary.banner_image;
    },
  },
  methods: {
    openVideoModal() {
      this.$modal.show("videoModal");
    },
    isVideo(link) {
      return link?.banner_link_url?.url?.includes("pix-videos/");
    },
    videoClass(link) {
      return this.isVideo(link)
        ? "banner__video"
        : "button--" + this.fontContrast;
    },
  },
  setup() {
    const img = useImage();

    const background = () => {
      let style = {};
      if (this.hasBackgroundImage) {
        style = {
          // Use @nuxt/image to download image during static build
          background: `no-repeat url(${img(
            this.slice.primary.banner_background.url
          )})`,
          backgroundSize: "cover",
          backgroundPosition: "bottom",
        };
      } else {
        style.backgroundColor = this.slice.primary.banner_background_color;
      }
      return style;
    };

    return background;
  },
};
</script>

<style lang="scss">
.banner {
  text-align: center;

  .text--normal p {
    margin: 16px 0 24px;
  }

  .banner__button-group {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;

    .button {
      margin: 5px 0;
    }
  }

  &.row-block {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-template-areas:
      "a a a a"
      "b b b b";
    grid-gap: 16px;

    font-family: $font-open-sans;
    padding: 81px 0;
    margin: 0 16px;

    & > * {
      max-width: 100%;
    }

    .row-block__main-content {
      grid-area: a;
    }

    .row-block__side-content {
      grid-area: b;
      display: none;
    }
  }
}

@include device-is("tablet") {
  .banner {
    padding: 80px 0;

    .banner__button-group {
      flex-direction: row;

      .button {
        width: initial;
        margin: 0 10px;
      }
    }

    &.row-block {
      margin: 0 32px;
      grid-template-columns: repeat(8, 1fr);
      grid-template-areas:
        "a a a a a a a a"
        "b b b b b b b b";
    }
  }
}

@include device-is("desktop") {
  .banner.row-block {
    margin: 0 32px;
    grid-template-columns: repeat(12, 1fr);
    grid-template-areas:
      "a a a a a a a a a a a a"
      ". . . b b b b b b . . .";

    .row-block__side-content {
      display: initial;
      width: 100%;
      box-sizing: border-box;
    }
  }
}

@include device-is("large-screen") {
  .banner.row-block {
    margin: 0 auto;
    grid-template-columns: 1.2fr repeat(12, 1fr) 1.2fr;
    grid-template-areas: ". a a a a a a . b b b b b .";
    grid-gap: 24px;

    .row-block__main-content {
      text-align: left;
      align-self: center;

      .banner__button-group {
        justify-content: flex-start;

        .button {
          margin: 0 20px 0 0;
        }
      }
    }
  }
}
</style>
