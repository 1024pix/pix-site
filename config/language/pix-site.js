import { config } from '../environment'

export const availableLocales = [
  {
    code: 'fr',
    file: 'fr.js',
    domain: config.domain.international,
    name: 'International FranÃ§ais',
    icon: 'globe-europe.svg',
  },
  {
    code: 'en-gb',
    file: 'en-gb.js',
    domain: config.domain.international,
    name: 'International English',
    icon: 'globe-europe.svg',
  },
  {
    code: 'fr-be',
    file: 'fr-be.js',
    domain: config.domain.international,
    name: 'FWB',
    icon: 'flag-be.svg',
  },
  {
    code: 'fr-fr',
    file: 'fr-fr.js',
    domain: config.domain.french,
    name: 'France',
    icon: 'flag-fr.svg',
  },
]

export const language = {
  menu: [
    {
      name: 'international',
      icon: 'globe-europe.svg',
      subtitle: null,
      children: [
        {
          name: 'french',
          localeCode: 'fr',
          icon: 'icon',
          subtitle: null,
        },
        {
          name: 'english',
          localeCode: 'en-gb',
          icon: 'icon',
          subtitle: null,
        },
      ],
    },
    {
      name: 'france',
      localeCode: 'fr-fr',
      icon: 'flag-fr.svg',
      subtitle: null,
    },
    {
      name: 'fwb-acronym',
      localeCode: 'fr-be',
      icon: 'flag-be.svg',
      subtitle: 'fwb',
    },
  ],
  locales: availableLocales,
  localesForCurrentSite: getLocalesForCurrentSite(),
}

export function getLocalesForCurrentSite() {
  if (config.isFrenchDomain) {
    return [{ code: 'fr-fr', file: 'fr-fr.js' }]
  }

  return availableLocales.filter((locale) => locale.code !== 'fr-fr')
}
