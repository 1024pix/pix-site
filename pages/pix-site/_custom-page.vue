<template>
  <div>
    <div v-if="type === FORM_PAGE">
      <form-page :content="document.data" />
    </div>
    <div v-if="type === SIMPLE_PAGE">
      <simple-page :content="document.data" />
    </div>
    <div v-if="type === SLICES_PAGE">
      <prismic-custom-slice-zone :slices="slices" />
    </div>
  </div>
</template>

<script>
import { documentFetcher, DOCUMENTS } from '~/services/document-fetcher'

export default {
  name: 'CustomPage',
  nuxtI18n: {
    paths: {
      fr: '/:uid',
      'fr-fr': '/:uid',
      'en-gb': '/:uid',
      'fr-be': '/:uid',
    },
  },
  async asyncData({ params, app, req, error, currentPagePath }) {
    try {
      const document = await documentFetcher(
        app.$prismic,
        app.i18n,
        req
      ).getPageByUid(params.uid)

      const meta = document.data.meta

      const latestNewsItems = await documentFetcher(
        app.$prismic,
        app.i18n,
        req
      ).findNewsItems({ page: 1, pageSize: 3 })
      return { currentPagePath, meta, document, latestNewsItems }
    } catch (e) {
      error({ statusCode: 404, message: 'Page not found' })
    }
  },
  data() {
    return {
      FORM_PAGE: DOCUMENTS.FORM_PAGE,
      SIMPLE_PAGE: DOCUMENTS.SIMPLE_PAGE,
      SLICES_PAGE: DOCUMENTS.SLICES_PAGE,
    }
  },
  head() {
    const meta = this.$getMeta(this.meta, this.currentPagePath, this.$prismic)

    return {
      meta,
      title: `${this.title} | Pix`,
    }
  },
  computed: {
    slices() {
      const rawDocumentSlices = this.document.data.body
      return rawDocumentSlices.map((slice, index) => {
        if (slice.slice_type === 'latest_news') {
          slice.primary.latest_news_items = this.latestNewsItems
        }
        return slice
      })
    },
    type() {
      return this.document.type
    },
    title() {
      return this.document.data.title[0].text
    },
  },
}
</script>

<style lang="scss"></style>
