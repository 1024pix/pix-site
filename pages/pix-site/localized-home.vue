<template>
  <div class="index">
    <prismic-custom-slice-zone :slices="slices" />
  </div>
</template>

<script>
import { documentFetcher } from '~/services/document-fetcher'

export default {
  name: 'LocalizedHome',
  nuxtI18n: {
    paths: {
      fr: '/',
      'fr-fr': '/',
      'en-gb': '/',
      'fr-be': '/',
    },
  },
  async asyncData({ app, req, error, currentPagePath }) {
    try {
      const document = await documentFetcher(
        app.$prismic,
        app.i18n,
        req
      ).getPageByUid('index-pix-site')

      const latestNewsItems = await documentFetcher(
        app.$prismic,
        app.i18n,
        req
      ).findNewsItems({ page: 1, pageSize: 3 })

      return {
        currentPagePath,
        meta: document.data.meta,
        document: document.data.body,
        title: document.data.title[0].text,
        latestNewsItems,
      }
    } catch (e) {
      console.error({ e })
      error({ statusCode: 404, message: 'Page not found' })
    }
  },
  head() {
    const meta = this.$getMeta(this.meta, this.currentPagePath, this.$prismic)
    return {
      title: `${this.title} | Pix`,
      meta,
    }
  },
  computed: {
    slices() {
      return this.document.map((slice, index) => {
        if (slice.slice_type === 'latest_news') {
          slice.primary.latest_news_items = this.latestNewsItems
        }
        return slice
      })
    },
  },
}
</script>
