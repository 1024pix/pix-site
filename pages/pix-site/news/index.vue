<template>
  <div class="news">
    <header class="page-header">
      <div class="container md padding-container">
        <h1 class="page-header__title">
          {{ $t('news-page-title') }}
        </h1>
      </div>
    </header>

    <main class="page-body">
      <h2 class="sr-only">
        {{ $t('news-page-title-level-two') }}
      </h2>
      <ul class="news__list">
        <template v-if="newsItems && newsItems.length">
          <news-item-card
            v-for="(item, index) in newsItems"
            :key="`item-${index}`"
            :slice="item.data"
            :uid="item.uid"
          />
        </template>
        <template v-else>
          <p>{{ $t('news-page-no-news') }}</p>
        </template>
      </ul>
    </main>
  </div>
</template>
<script>
import { documentFetcher } from '~/services/document-fetcher'

export default {
  name: 'Index',
  nuxtI18n: {
    paths: {
      fr: '/actualites',
      'fr-fr': '/actualites',
      'en-gb': '/news',
    },
  },
  async asyncData({ app, error, req }) {
    try {
      const newsItems = await documentFetcher(
        app.$prismic,
        app.i18n,
        req
      ).findNewsItems()
      return { newsItems }
    } catch (e) {
      // eslint-disable-next-line no-console
      console.error({ e })
      error({ statusCode: 404, message: 'Page not found' })
    }
  },
  head() {
    const pageTitle = this.$t('page-titles.news')
    return {
      title: pageTitle,
    }
  },
}
</script>

<style lang="scss">
.news {
  h1,
  h2,
  h3,
  h4,
  p {
    font-family: $font-open-sans;
  }

  &__list {
    max-width: 1140px;
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    padding-left: 0;

    @include device-is('tablet') {
      flex-wrap: wrap;
      flex-direction: row;
      justify-content: space-evenly;
    }

    li {
      padding-left: 0;

      &::before {
        content: none;
      }
    }
  }
}
</style>
