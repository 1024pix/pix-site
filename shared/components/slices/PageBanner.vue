<template>
  <div :id="`slice-${indexForId}`" :style="[background]">
    <section class="banner row-block">
      <div class="row-block__main-content">
        <prismic-rich-text
          :field="slice.primary.banner_title"
          class="title--big"
          :class="`title--${slice.primary.banner_font_contrast}`"
        />
        <prismic-rich-text
          :field="slice.primary.banner_content"
          class="text--normal"
          :class="`text--${slice.primary.banner_font_contrast}`"
        />
        <slot></slot>
        <div class="banner__button-group">
          <div v-for="(link, index) in slice.items" :key="`item-${index}`">
            <nuxt-link
              v-if="!isVideo(link)"
              :to="link.banner_link_url.url"
              class="button"
              :class="videoClass(link)"
            >
              {{ link.banner_link_text }}
            </nuxt-link>
            <template v-if="isVideo(link)">
              <button
                class="button button-video"
                :class="`button--${fontContrast}`"
                @click="toggleModal"
              >
                <fa icon="play-circle" />
                {{ link.banner_link_text }}
              </button>
              <modal v-if="showModal" @close-modal="toggleModal">
                <media-player :video-url="link.banner_link_url.url" />
              </modal>
            </template>
          </div>
        </div>
      </div>
      <pix-image
        v-if="hasImage"
        :field="slice.primary.banner_image"
        class="row-block__side-content"
      />
    </section>
  </div>
</template>

<script setup>
const img = useImage();

const props = defineProps({
  slice: {
    type: Object,
    default: null,
  },
  indexForId: {
    type: Number,
    default: 0,
  },
});

/* Background */
const hasBackgroundImage =
  props.slice.primary.banner_background &&
  props.slice.primary.banner_background.url;

let background = {};
if (hasBackgroundImage) {
  background = {
    // Use @nuxt/image to download image during static build
    background: `no-repeat url(${img(
      props.slice.primary.banner_background.url
    )})`,
    backgroundSize: "cover",
    backgroundPosition: "bottom",
  };
} else {
  background.backgroundColor = props.slice.primary.banner_background_color;
}

/* Image */
const hasImage =
  props.slice.primary.banner_image && props.slice.primary.banner_image.url;

/* Video */
function isVideo(link) {
  return link?.banner_link_url?.url?.includes("pix-videos/");
}

function videoClass(link) {
  return isVideo(link)
    ? "banner__video"
    : "button--" + props.slice.primary.banner_font_contrast;
}

/* Modal */
const showModal = ref(false);

function toggleModal() {
  showModal.value = !showModal.value;
}
</script>

<style lang="scss">
.banner {
  text-align: center;

  .text--normal p {
    margin: 16px 0 24px;
  }

  .banner__button-group {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;

    .button {
      margin: 5px 0;
    }
  }

  &.row-block {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-template-areas:
      "a a a a"
      "b b b b";
    grid-gap: 16px;

    font-family: $font-open-sans;
    padding: 81px 0;
    margin: 0 16px;

    & > * {
      max-width: 100%;
    }

    .row-block__main-content {
      grid-area: a;
    }

    .row-block__side-content {
      grid-area: b;
      display: none;
    }
  }
}

@include device-is("tablet") {
  .banner {
    padding: 80px 0;

    .banner__button-group {
      flex-direction: row;

      .button {
        width: initial;
        margin: 0 10px;
      }
    }

    &.row-block {
      margin: 0 32px;
      grid-template-columns: repeat(8, 1fr);
      grid-template-areas:
        "a a a a a a a a"
        "b b b b b b b b";
    }
  }
}

@include device-is("desktop") {
  .banner.row-block {
    margin: 0 32px;
    grid-template-columns: repeat(12, 1fr);
    grid-template-areas:
      "a a a a a a a a a a a a"
      ". . . b b b b b b . . .";

    .row-block__side-content {
      display: initial;
      width: 100%;
      box-sizing: border-box;
    }
  }
}

@include device-is("large-screen") {
  .banner.row-block {
    margin: 0 auto;
    grid-template-columns: 1.2fr repeat(12, 1fr) 1.2fr;
    grid-template-areas: ". a a a a a a . b b b b b .";
    grid-gap: 24px;

    .row-block__main-content {
      text-align: left;
      align-self: center;

      .banner__button-group {
        justify-content: flex-start;

        .button {
          margin: 0 20px 0 0;
        }
      }
    }
  }
}
</style>
