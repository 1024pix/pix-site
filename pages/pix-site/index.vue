<template>
  <client-only v-if="shouldDisplayLocaleChoice">
    <main class="home">
      <div class="locale-choice">
        <h1>
          <pix-image
            class="logo-pix"
            :field="{ url: '/images/logo-pix-blanc.svg', alt: 'Pix' }"
          />
        </h1>
        <locale-link
          v-for="locale in locales"
          :key="locale.code"
          class="locale-choice__link"
          :locale="locale"
        />
        <pix-image class="planet" :field="{ url: '/images/planet.svg' }" />
      </div>
    </main>
  </client-only>
</template>

<script>
import { language } from '~/config/language'

export default {
  layout: 'empty',
  nuxtI18n: false,
  data() {
    return {
      locales: language.locales,
      shouldDisplayLocaleChoice: false,
    }
  },
  mounted() {
    const chosenLocale = this.getLocaleFromCookie()
    if (chosenLocale) {
      return this.$router.push(`/${chosenLocale}/`)
    }
    this.shouldDisplayLocaleChoice = true
  },
  methods: {
    getLocaleFromCookie() {
      const localeCookie = document.cookie
        .split('; ')
        .find((item) => item.startsWith('locale'))
      if (!localeCookie) return null

      const chosenLocale = localeCookie.split('=')[1]

      const currentLocales = language.localesForCurrentSite.map(
        ({ code }) => code
      )
      if (!currentLocales.includes(chosenLocale)) return null

      return chosenLocale
    },
  },
}
</script>

<style lang="scss" scoped>
.home {
  width: 100%;
  min-width: 0;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 5rem 1rem 0;
  background: url('/images/stars.svg') repeat, $default-gradient;

  @include device-is('large-mobile') {
    padding: 0;
  }
}

.locale-choice {
  padding: 3rem 0;
}

.logo-pix {
  margin: 0 auto 5vh;
  display: block;
  width: 11rem;
  max-width: 100%;
}

.locale-choice__link + .locale-choice__link {
  margin-top: 1rem;
}

.planet {
  margin: 5vh auto 0;
  display: block;
  width: max(16vw, 90%);
}

@include device-is('large-mobile') {
  .locale-choice {
    position: relative;
  }

  .planet {
    position: absolute;
    top: 50%;
    right: calc(100% + 4vw);
    max-width: unset;
    margin: 0;
    transform: translateY(-50%);
  }
}
</style>
