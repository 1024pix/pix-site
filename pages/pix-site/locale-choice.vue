<template>
  <client-only>
    <div class="locale-choice">
      <h1>
        <pix-image
          class="logo-pix"
          :field="{ url: '/images/logo-pix-blanc.svg', alt: 'Pix' }"
        />
      </h1>
      <LocaleLink
        v-for="locale in locales"
        :key="locale.code"
        class="locale-choice__link"
        :locale="locale"
      />
      <pix-image class="planet" :field="{ url: '/images/planet.svg' }" />
    </div>
  </client-only>
</template>

<script>
import { language } from '~/config/language'
export default {
  layout: 'home',
  nuxtI18n: false,
  data() {
    return {
      locales: language.locales,
    }
  },
}
</script>

<style lang="scss" scoped>
.locale-choice {
  padding: 3rem 0;
}

.logo-pix {
  margin: 0 auto 5vh;
  display: block;
  width: 11rem;
  max-width: 100%;
}

.locale-choice__link + .locale-choice__link {
  margin-top: 1rem;
}

.planet {
  margin: 5vh auto 0;
  display: block;
  width: max(16vw, 90%);
}

@include device-is('large-mobile') {
  .locale-choice {
    position: relative;
  }

  .planet {
    position: absolute;
    top: 50%;
    right: calc(100% + 4vw);
    max-width: unset;
    margin: 0;
    transform: translateY(-50%);
  }
}
</style>
