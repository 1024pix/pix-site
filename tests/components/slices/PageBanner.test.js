import { shallowMount } from '@vue/test-utils'
import VModal from 'vue-js-modal'
import { createLocalVue } from '../../pages/pix-site/utils'
import PageBanner from '~/components/slices/PageBanner'

const localVue = createLocalVue()
localVue.use(VModal)

describe('Slice: Page Banner', () => {
  let component

  it('should render component', () => {
    // given & when
    component = shallowMount(PageBanner, {
      localVue,
      stubs: {
        'prismic-rich-text': true,
        fa: true,
        'media-player': true,
      },
      propsData: {
        slice: {
          primary: {
            banner_font_contrast: '',
            title: '',
            textContent: '',
            hasImage: false,
            hasBackgroundImage: false,
            imageUrl: undefined,
          },
          items: [{ banner_link_url: { url: 'pix-videos/' } }],
        },
      },
    })

    // then
    const result = component.html()
    expect(result).toMatchSnapshot()
  })
})
