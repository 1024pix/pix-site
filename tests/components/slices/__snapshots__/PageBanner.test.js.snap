// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slice: Page Banner should render component 1`] = `
"<div id="slice-0">
  <section class="banner row-block">
    <div class="row-block__main-content">
      <prismic-rich-text-stub class="title--big title--"></prismic-rich-text-stub>
      <prismic-rich-text-stub class="text--normal text--"></prismic-rich-text-stub>
      <div class="banner__button-group">
        <div>
          <!----> <button class="button button-video button--">
            <fa-stub icon="play-circle"></fa-stub>

          </button>
          <modal-stub name="videoModal" resizeedges="r,br,b,bl,l,tl,t,tr" centerresize="true" resizeindicator="true" overlaytransition="vm-transition--overlay" transition="vm-transition--modal" clicktoclose="true" classes="" minwidth="0" minheight="0" maxwidth="9007199254740991" maxheight="9007199254740991" width="600" height="auto" shiftx="0.5" shifty="0.5">
            <media-player-stub video-url="pix-videos/"></media-player-stub>
          </modal-stub>
        </div>
      </div>
    </div>
    <!---->
  </section>
</div>"
`;
